nxdrive/gui/view.py

```diff
@@ -439,10 +510,26 @@ def setData(self, index: QModelIndex, value: Any, role: int = None) -> None:
         self.sessions[index.row()][key] = value
         self.dataChanged.emit(index, index, [role])
 
+    def add_session(self, parent: QModelIndex, n_session: Dict[str, Any]) -> None:
+        """Add a session to existing list."""
+        self.beginInsertRows(parent, self.rowCount(), self.rowCount())
+        self.sessions.append(n_session)
+        self.endInsertRows()
+
+    def edit_session(self, row: int, n_session: Dict[str, Any]) -> None:
+        """Replace an existing session with *n_session*."""
+        idx = self.index(row, 0)
+        self.sessions[row] = n_session
+        self.dataChanged.emit(idx, idx, self.roleNames())
+
     @pyqtProperty("int", notify=sessionChanged)
     def count(self) -> int:
         return self.rowCount()
 
+    @pyqtProperty("bool", notify=sessionChanged)
+    def is_max(self) -> bool:
```

Comment:
```suggestion
    def is_full(self) -> bool:
```

Or `has_reached_max_sesions_count`.